### PR TITLE
Update maintenance_tasks README

### PR DESCRIPTION
**Pull request by noahgibbs**
... to explain more clearly what a maintenance task is and when you'd use it.

This is updated with changes from pairing with @adrianna-chang-shopify.

### CODE REVIEW

**Review by lastgabs — COMMENTED** [review]
Thanks for improving the docs @noahgibbs and @adrianna-chang-shopify !

Added small comments and questions. Let me know if you prefer me to add them as "commit suggestions" next time.

File: README.md [hunk 1]
```diff
@@ -2,8 +2,35 @@
...
+
+Maintenance tasks aren't meant to happen on a regular basis. They're used as needed, or as one-offs. Normally maintenance tasks are ephemeral, so they are used briefly and then deleted.
+
+The Rails engine has a web-based UI for listing maintenance tasks, seeing their status, and starting, pausing and restarting them.
+
 [![Link to demo video](static/demo.png)](https://www.youtube.com/watch?v=BTuvTQxlFzs)
 
+## Should I Use Maintenance Tasks?
+
+Maintenance Tasks have a limited, specific job UI. While the engine can used to provide a user interface for other data changes, such as data changes for support requests for example, we recommend you use regular application code or [Rails administration engines][rails-admin-engines] for those use cases instead. These inevitably require more flexibility than this engine will be able to provide.
```
Comment (adrianna-chang-shopify) [inline README.md:18]:
```suggestion
Maintenance Tasks have a limited, specific job UI. While the engine can used to provide a user interface for other data changes, such as data changes for support requests, we recommend you use regular application code or [Rails administration engines][rails-admin-engines] for those use cases instead. These inevitably require more flexibility than this engine will be able to provide.
```

File: README.md [hunk 2]
```diff
@@ -2,8 +2,35 @@
...
+
+If your task updates schema instead of data, use a migration instead of a Maintenance Task.
+
+If your task happens regularly, consider Active Jobs with a scheduler or cron, [job-iteration jobs](https://github.com/shopify/job-iteration) and/or [custom rails_admin UIs][rails-admin-enginers] instead of a Maintenance Task. Maintenance Tasks should be ephemeral, to suit their intentionally limited UI.
+
+To create seed data for a new application, use Rails db/seeds.rb.
+
+If your application can't handle a half-completed migration, Maintenance Tasks are probably the wrong tool. Remember that they are intentionally pausable and can be cancelled halfway.
+
+[rails-admin-engines]: https://github.com/railsadminteam/rails_admin
```
Comment (adrianna-chang-shopify) [inline README.md:32]: As discussed, let's link to the Ruby Toolbox here instead!
https://www.ruby-toolbox.com/categories/rails_admin_interfaces

**Review by adrianna-chang-shopify — COMMENTED** [review]
Looking much better! A couple more suggestions 😄

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (noahgibbs) [reply README.md]: Fixed, thanks!

**Pull request merged**